DOCS-6806 - Connect to Datadog over GCP Private Service Connect (#20976)
DataDog/documentation: content/en/agent/guide/gcp-private-service-connect.md

---
title: Connect to Datadog over GCP Private Service Connect
kind: guide
further_reading:
    - link: '/integrations/google_cloud_platform/'
      tag: 'Documentation'
      text: 'Datadog-Google Cloud Platform Integration'
    - link: '/agent/guide/private-link'
      tag: 'Documentation'
      text: 'Connect to Datadog over AWS PrivateLink'
---

{{% site-region region="us,us3,eu,gov,ap1" %}}
<div class="alert alert-warning">This feature is not supported for the selected Datadog site.</div>
{{% /site-region %}}

{{% site-region region="us5" %}}

> [Private Service Connect][1] (PSC) allows you to connect directly to Datadog without using the public internet.

Reviewer: It's not clear to me what entity is connecting to Datadog, but perhaps its clear to users of GCP.

Author: good point, going to modify this a little bit

```suggestion
[Private Service Connect][1] (PSC) allows you to send telemetry to Datadog without using the public internet.
```

Datadog exposes some of its data intake services in Google Cloud Platform as PSC [_published services_][2], as seen in the [table of published services](#published-services). 

You can configure a GCP PSC endpoint to expose a private IP address for each Datadog intake service; this IP address routes traffic to the Datadog backend. You can then configures a GCP [_Private DNS Zone_][3] to override the DNS names corresponding to the products for each endpoint that is consumed. 

{{< img src="agent/guide/psc/gcp-psc-overview-1.png" alt="GCP Private Service Connect schema. On the left, a 'Customer VPC' box contains Datadog Agents sending data to a PSC endpoint. On the right, a 'Datadog VPC' box contains a service attachment in communication with Datadog services. The PSC endpoint in the 'Customer VPC' box connects to the service attachment in the 'Datadog VPC' box through the GCP Private Backbone. " >}}

## Setup

### Connect an endpoint

1. In your GCP console, navigate to **Network services** > **Private Service Connect**.
2. Go to the **Endpoints** section. Click on **Connect endpoint**.

>    {{< img src="agent/guide/psc/connect-endpoint.png" alt="Screenshot of a 'Connect endpoint' page in the GCP console" >}}

Reviewer: I agree with your point about needing the screenshot of the external site for the instructions to make sense.

   **Note**: Datadog exposes PSC producer endpoints from the `us-central1` region. These endpoints support global access, allowing services to connect from any region. However, the forwarding rule must be created in the `us-central1` region.

3. Click **Add endpoint**. Verify that your status is _Accepted_. Take note of the IP address, as this is used in the next section.
   {{< img src="agent/guide/psc/connect-endpoint-success.png" alt="Screenshot of a success message after adding an endpoint in the GCP console. Includes an IP address" >}}

### Create a DNS zone
1. In your GCP console, navigate to **Network services** > **Cloud DNS**.
2. Click on **Create zone**.
   {{< img src="agent/guide/psc/create-a-dns-zone.png" alt="Screenshot of a 'Create a DNS zone' page in the GCP console" >}}

   - Under **Zone type**, select _Private_.
   - For **Zone name**, enter a descriptive name for your zone.
   - For **DNS name**, enter the _private DNS name_ that corresponds to the Datadog intake service that you want to use. You can find your DNS name in the [table of published services](#published-services).

> 3. Next, create an `A` record that points to the endpoint IP. On the _Zone details_ page of the zone you created, click on **Add record set**.

Reviewer: Does the user need to know what an `A` record is?

Author: although it might help the user to conceptually understand what they're about to do (an `A` record maps a hostname to an IPv4 address), in this situation the user only really needs a functional definition: an `A` record is one where you select `A` from the dropdown under "resource record type". this direction is explicit in the instructions that follow.

also, `A` records should be fairly common knowledge in the devops world. of course, relying on a user's assumed knowledge is a little dangerous—but in this case where they don't *need* the definition to successfully complete this configuration, i think it's fine to omit it.

   {{< img src="agent/guide/psc/create-record.png" alt="Screenshot of the 'Create record set' page in the GCP console." >}}

   - For **DNS name**, leave the field unmodified.
   - For **Resource record type**, select `A`.
   - Under **IPv4 Address**, enter the IP address that was displayed at the end of the previous section.

### Additional required steps for metrics and traces

> To set up DNS for the metrics or traces endpoint (`metrics.agent5.us5.datadoghq.com` or `trace.agent.us5.datadoghq.com`), additional steps are required.

Reviewer: The sentence would be stronger in active voice, but it is understandable in passive.

Author: ```suggestion
To set up DNS for the metrics or traces endpoint (`metrics.agent5.us5.datadoghq.com` or `trace.agent.us5.datadoghq.com`), you must create additional `A` records.
```
